### PR TITLE
Reverse tab order in topics with tabs (2.x first)

### LINKED ISSUE #4089: Reverse tab order in topics with tabs (2.x first)
There are a number of topics with the tab order 1.x-2.x, but I understand it was settled that the topics should render the tab with the latest release info, thus they should be 2.x first. I don't have a list. I patched one and then realized it wasn't a one-off when I saw others that need the same work. Seems that all of them should be addressed on one PR.

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #4089

* Also touches just a few sample code layout issues with spacing and some unescaped asterisks.
* Does **not** change the order of tabs for the *fundamentals/hosting* topic that I'm working on. I have that on a local branch with the tab order fixed, so let's allow that topic to update on its own upcoming PR.

### CODE REVIEW

File: aspnetcore/security/authentication/social/microsoft-logins.md [hunk 1]
```diff
@@ -74,18 +74,6 @@ The project template used in this tutorial ensures that
...
-app.UseMicrosoftAccountAuthentication(new MicrosoftAccountOptions()
-{
-    ClientId = Configuration["Authentication:Microsoft:ApplicationId"],
-    ClientSecret = Configuration["Authentication:Microsoft:Password"]
-});
-```
-
 # [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
 Add the Microsoft Account middleware in the `ConfigureServices` method in *Startup.cs* file:
```
Comment (scottaddie) [inline aspnetcore/security/authentication/social/microsoft-logins.md]: - middleware --> service
- The `AddMicrosoft` method below should be `AddMicrosoftAccount`

File: aspnetcore/security/authentication/social/twitter-logins.md [hunk 2]
```diff
@@ -55,18 +55,6 @@ The project template used in this tutorial ensures that [Microsoft.AspNetCore.Au
...
-app.UseTwitterAuthentication(new TwitterOptions()
-{
-    ConsumerKey = Configuration["Authentication:Twitter:ConsumerKey"],
-    ConsumerSecret = Configuration["Authentication:Twitter:ConsumerSecret"]
-});
-```
-
 # [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
 Add the Twitter middleware in the `ConfigureServices` method in *Startup.cs* file:
```
Comment (scottaddie) [inline aspnetcore/security/authentication/social/twitter-logins.md]: middleware --> service

File: aspnetcore/security/authentication/social/google-logins.md [hunk 3]
```diff
@@ -91,18 +91,6 @@ The project template used in this tutorial ensures that [Microsoft.AspNetCore.Au
...
-app.UseGoogleAuthentication(new GoogleOptions()
-{
-    ClientId = Configuration["Authentication:Google:ClientId"],
-    ClientSecret = Configuration["Authentication:Google:ClientSecret"]
-});
-```
-
 # [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
 Add the Google middleware in the `ConfigureServices` method in *Startup.cs* file:
```
Comment (scottaddie) [inline aspnetcore/security/authentication/social/google-logins.md]: middleware --> service

File: aspnetcore/security/authentication/social/facebook-logins.md [hunk 4]
```diff
@@ -69,18 +69,6 @@ The project template used in this tutorial ensures that [Microsoft.AspNetCore.Au
...
-app.UseFacebookAuthentication(new FacebookOptions()
-{
-    AppId = Configuration["Authentication:Facebook:AppId"],
-    AppSecret = Configuration["Authentication:Facebook:AppSecret"]
-});
-```
-
 # [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
 Add the Facebook middleware in the `ConfigureServices` method in the *Startup.cs* file:
```
Comment (scottaddie) [inline aspnetcore/security/authentication/social/facebook-logins.md]: middleware --> service

File: aspnetcore/publishing/web-publishing-vs.md [hunk 5]
```diff
@@ -21,46 +21,45 @@ This article focuses on using Visual Studio 2017 to create publish profiles. The
 
 The following *.csproj* file was created with the command `dotnet new mvc`:
 
-# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
```
Comment (scottaddie) [inline aspnetcore/publishing/web-publishing-vs.md]: The 2.x *.csproj* file produced should be the following:

```xml
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netcoreapp2.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.All" Version="2.0.0" />
  </ItemGroup>

  <ItemGroup>
    <DotNetCliToolReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Tools" Version="2.0.0" />
  </ItemGroup>

</Project>
```

File: aspnetcore/fundamentals/servers/kestrel.md [hunk 6]
```diff
@@ -212,12 +208,57 @@ You can configure endpoints by calling the `UseUrls` method or using the `urls`
 
 If you use IIS, the URL bindings for IIS override any bindings that you set by calling either `Listen` or `UseUrls`. For more information, see [Introduction to ASP.NET Core Module](aspnet-core-module.md).
 
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+
+By default ASP.NET Core binds to `http://localhost:5000`. You can configure URL prefixes and ports for Kestrel to listen on by using the `UseUrls` extension method, the `urls` command-line argument, or the ASP.NET Core configuration system. For more information about these methods, see [Hosting](../../fundamentals/hosting.md). For information about how URL binding works when you use IIS as a reverse proxy, see [ASP.NET Core Module](aspnet-core-module.md).
```
Comment (scottaddie) [inline aspnetcore/fundamentals/servers/kestrel.md]: - Add a comma after "By default".
- Remove hyphen from "command-line".

File: aspnetcore/fundamentals/servers/aspnet-core-module.md [hunk 7]
```diff
@@ -57,46 +57,46 @@ The ASP.NET Core Module has to be installed in IIS on your servers and in IIS Ex
...
-[!code-csharp[](aspnet-core-module/sample/Program.cs?name=snippet_Main&highlight=12)]
+---
+
+### Call UseIISIntegration
 
 # [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
 The `UseIISIntegration` extension method on [`WebHostBuilder`](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.hosting.webhostbuilder) is called automatically when you run with IIS.
 
 If you aren't using one of the ASP.NET Core metapackages and haven't installed the  `Microsoft.AspNetCore.Server.IISIntegration` package, you get a runtime error. If you call `UseIISIntegration` explicitly, you get a compile time error if the package isn't installed.
```
Comment (scottaddie) [inline aspnetcore/fundamentals/servers/aspnet-core-module.md]: Remove the extra space before `Microsoft.AspNetCore.Server.IISIntegration`.

File: aspnetcore/fundamentals/servers/aspnet-core-module.md [hunk 8]
```diff
@@ -57,46 +57,46 @@ The ASP.NET Core Module has to be installed in IIS on your servers and in IIS Ex
...
 
----
-
-### Call UseIISIntegration
-
 # [ASP.NET Core 1.x](#tab/aspnetcore1x)
 
-In your application's `Main` method, call the `UseIISIntegration` extension method on [`WebHostBuilder`](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.hosting.webhostbuilder). 
+In your application, install [Microsoft.AspNetCore.Server.IISIntegration](https://www.nuget.org/packages/Microsoft.AspNetCore.Server.IISIntegration/). The `IISIntegration` package  is an interoperability pack that reads environment variables broadcast by ANCM to set up your app. The environment variables provide configuration information, such as the port to listen on.
```
Comment (scottaddie) [inline aspnetcore/fundamentals/servers/aspnet-core-module.md]: Remove the extra space before "is an interoperability".

File: aspnetcore/security/data-protection/configuration/overview.md [hunk 9]
```diff
@@ -146,47 +145,47 @@ The developer can manually specify an implementation if desired via a call to Us
 
 ### Specifying custom managed algorithms
 
-# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
-To specify custom managed algorithms, create a ManagedAuthenticatedEncryptionSettings instance that points to the implementation types.
+To specify custom managed algorithms, create a ManagedAuthenticatedEncryptorConfiguration instance that points to the implementation types.
```
Comment (scottaddie) [inline aspnetcore/security/data-protection/configuration/overview.md]: Enclose ManagedAuthenticatedEncryptorConfiguration in back ticks.

File: aspnetcore/security/data-protection/configuration/overview.md [hunk 10]
```diff
@@ -146,47 +145,47 @@ The developer can manually specify an implementation if desired via a call to Us
...
+        // a type that subclasses KeyedHashAlgorithm
+        ValidationAlgorithmType = typeof(HMACSHA256)
+    });
+```
 
-# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
 
-To specify custom managed algorithms, create a ManagedAuthenticatedEncryptorConfiguration instance that points to the implementation types.
+To specify custom managed algorithms, create a ManagedAuthenticatedEncryptionSettings instance that points to the implementation types.
```
Comment (scottaddie) [inline aspnetcore/security/data-protection/configuration/overview.md]: Enclose ManagedAuthenticatedEncryptorConfiguration in back ticks.

File: aspnetcore/security/data-protection/configuration/overview.md [hunk 11]
```diff
@@ -146,47 +145,47 @@ The developer can manually specify an implementation if desired via a call to Us
...
-   ```
+        // a type that subclasses KeyedHashAlgorithm
+        ValidationAlgorithmType = typeof(HMACSHA256)
+    });
+```
 
 ---
 
-Generally the *Type properties must point to concrete, instantiable (via a public parameterless ctor) implementations of SymmetricAlgorithm and KeyedHashAlgorithm, though the system special-cases some values like typeof(Aes) for convenience.
+Generally the \*Type properties must point to concrete, instantiable (via a public parameterless ctor) implementations of SymmetricAlgorithm and KeyedHashAlgorithm, though the system special-cases some values like typeof(Aes) for convenience.
```
Comment (scottaddie) [inline aspnetcore/security/data-protection/configuration/overview.md]: Add a comma after "Generally".

File: aspnetcore/security/data-protection/configuration/overview.md [hunk 12]
```diff
@@ -195,87 +194,86 @@ Generally the *Type properties must point to concrete, instantiable (via a publi
 
 ### Specifying custom Windows CNG algorithms
 
-# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
-To specify a custom Windows CNG algorithm using CBC-mode encryption + HMAC validation, create a CngCbcAuthenticatedEncryptionSettings instance that contains the algorithmic information.
+To specify a custom Windows CNG algorithm using CBC-mode encryption + HMAC validation, create a CngCbcAuthenticatedEncryptorConfiguration instance that contains the algorithmic information.
```
Comment (scottaddie) [inline aspnetcore/security/data-protection/configuration/overview.md]: Enclose CngCbcAuthenticatedEncryptorConfiguration in back ticks.

File: aspnetcore/security/data-protection/configuration/overview.md [hunk 13]
```diff
@@ -195,87 +194,86 @@ Generally the *Type properties must point to concrete, instantiable (via a publi
...
+        HashAlgorithm = "SHA256",
+        HashAlgorithmProvider = null
+    });
+```
 
-# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
 
-To specify a custom Windows CNG algorithm using CBC-mode encryption + HMAC validation, create a CngCbcAuthenticatedEncryptorConfiguration instance that contains the algorithmic information.
+To specify a custom Windows CNG algorithm using CBC-mode encryption + HMAC validation, create a CngCbcAuthenticatedEncryptionSettings instance that contains the algorithmic information.
```
Comment (scottaddie) [inline aspnetcore/security/data-protection/configuration/overview.md]: Enclose CngCbcAuthenticatedEncryptionSettings in back ticks.

File: aspnetcore/security/data-protection/configuration/overview.md [hunk 14]
```diff
@@ -195,87 +194,86 @@ Generally the *Type properties must point to concrete, instantiable (via a publi
...
+> The symmetric block cipher algorithm must have a key length of ≥ 128 bits and a block size of ≥ 64 bits, and it must support CBC-mode encryption with PKCS #7 padding. The hash algorithm must have a digest size of >= 128 bits and must support being opened with the BCRYPT_ALG_HANDLE_HMAC_FLAG flag. The \*Provider properties can be set to null to use the default provider for the specified algorithm. See the [BCryptOpenAlgorithmProvider](https://msdn.microsoft.com/library/windows/desktop/aa375479(v=vs.85).aspx) documentation for more information.
 
-# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
-To specify a custom Windows CNG algorithm using Galois/Counter Mode encryption + validation, create a CngGcmAuthenticatedEncryptionSettings instance that contains the algorithmic information.
+To specify a custom Windows CNG algorithm using Galois/Counter Mode encryption + validation, create a CngGcmAuthenticatedEncryptorConfiguration instance that contains the algorithmic information.
```
Comment (scottaddie) [inline aspnetcore/security/data-protection/configuration/overview.md]: Enclose CngGcmAuthenticatedEncryptorConfiguration in back ticks.

File: aspnetcore/security/data-protection/configuration/overview.md [hunk 15]
```diff
@@ -195,87 +194,86 @@ Generally the *Type properties must point to concrete, instantiable (via a publi
...
+
+        // specified in bits
+        EncryptionAlgorithmKeySize = 256
+    });
+```
 
-To specify a custom Windows CNG algorithm using Galois/Counter Mode encryption + validation, create a CngGcmAuthenticatedEncryptorConfiguration instance that contains the algorithmic information.
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+
+To specify a custom Windows CNG algorithm using Galois/Counter Mode encryption + validation, create a CngGcmAuthenticatedEncryptionSettings instance that contains the algorithmic information.
```
Comment (scottaddie) [inline aspnetcore/security/data-protection/configuration/overview.md]: Enclose CngGcmAuthenticatedEncryptionSettings in back ticks.

**Comment by scottaddie** [comment]
@guardrex Was this closed on accident? Just wondering if I should submit a PR to make some of my proposed changes.

**Comment by guardrex** [comment]
Not an accident ... I'm a bit under-the-:gun: on time, so I won't be able address these requested changes, so I thought it best to just close it. It really was just a passing notion when I was looking at one of the topics for updates to the hosting topic to complete the work for https://github.com/aspnet/Docs/issues/1870 and noticed that tab order hadn't been fixed.

**Comment by scottaddie** [comment]
@guardrex Understandable. If you merge it as is, I'll take care of the requested changes in a different PR.

**Comment by guardrex** [comment]
kk ... that's cool. The one tho ... "command-line" (dashed). As an adjective, it's cool.

**Pull request merged**